tests/Test/AbstractConfigTestCaseTest.php

```diff
@@ -70,6 +71,27 @@ public function testRuleSetsAreHandled()
         }
     }
 
+    public function testRuleSetsMustBeOrderedAsTheyAppearInRuleSetClass()
```

Comment:
ruleset in config must be ordered same as in ruleset class?
why??

Reply:
Set definitions are processed in the order they appear. For example, regarding the rule `visibility_required`, if you use:

```php
$config->setRules([
    '@PHP71Migration' => true,
    '@PSR2' => true,
]);
```
The `fix` run will **not** fix visibility for `const` because `@PSR2` resets the rule to the default value which doesn't fix constants, and this is likely not what the user expects.

I'll add the _why_ in the test name.